doc/ps_design_interactions_among_master_worker_ps.md

```diff
@@ -0,0 +1,54 @@
+# ElasticDL Parameter Server Design Doc
+## Parameter Server
+
+
+## Interactions among Master, PS and Worker
+
+The following events involve interactions among the master, workers and PS:
+
+* The master starts PS.
+* Initialization of parameters in PS.
+* Relaunch of PS.
+* Workers get model parameters from PS.
+* Workers push gradients to PS.
+* PS reports submodel version to the master.
+* The master tells PS to save checkpoint.
+
+
+### The master starts PS
+When an ElasticDL task starts, `master.main` is responsible for starting PS. After starting PS, `master.main` starts the master and workers, and tells them the endpoints of PS.
+
+### Initialization of parameters in PS
+PS does not have any model variable and mdoel meta info after starting. Model meta info includes dimension of embedding layers, initialization methods of embedding vectors, initialization methods of slot variables in optimizer.
+
+There are two ways for PS to get model variables and model meta info, one is to read from a checkpoint file, one is to obtain them from workers.
+
+When `master.main` starts PS, `master.main` decides how to initialize PS. If `master.main` passes an argument specifying the checkpoint file name to PS, PS reads from the checkpoint. Otherwise, PS does nothing but waiting for the first `get_model` from worker. In the reponse of `get_model` call, PS tells the worker to initialize model, and report model variables and meta info to the PS.
```

Comment:
What about meta info when PS reads from the checkpoint? The checkpoint should have initialization methods, etc?

Reply:
Yes, we save meta info into checkpoint file.